### PR TITLE
feat(HardwareDetails): add card with compatibles

### LINKED ISSUE #694: Add card with hardware compatibles in Hardware Details
Show the same information shown in test details in the field "Hardware" in a separate card in the Hardware Details page

### PR DESCRIPTION
**Pull request by murilx**
Closes #694 

## Visual reference
image

### CODE REVIEW

File: backend/kernelCI_app/views/hardwareDetailsView.py [hunk 1]
```diff
@@ -350,6 +351,9 @@ def sanitize_records(self, records, trees: List, is_all_selected: bool):
 
             self._assign_default_record_values(record)
 
+            if record["environment_compatible"] is not None:
+                compatibles.update(record["environment_compatible"])
```
Comment (Francisco2002) [inline backend/kernelCI_app/views/hardwareDetailsView.py:355]: nice find!!!

File: dashboard/src/components/Cards/CompatibleHardware.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}: ICompatible): JSX.Element => {
+  const compatiblesSorted = useMemo(() => compatibles.sort(), [compatibles]);
+
+  return (
+    <BaseCard
+      title={title}
+      className="mb-0"
+      content={
+        <div className="flex flex-row flex-wrap gap-4 p-4">
+          {compatiblesSorted?.map(compatible => (
```
Comment (Francisco2002) [inline dashboard/src/components/Cards/CompatibleHardware.tsx:24]: `compatibleSorted` isn't undefined, so you don't need this `?` before access the `map` method

File: dashboard/src/pages/hardwareDetails/HardwareDetails.tsx [hunk 3]
```diff
@@ -223,6 +225,12 @@ function HardwareDetails(): JSX.Element {
           selectedIndexes={treeIndexes}
           updateTreeFilters={updateTreeFilters}
         />
+        <div className="mt-5">
+          <MemoizedCompatibleHardware
+            title={<FormattedMessage id="hardwareDetails.compatibles" />}
+            compatibles={data?.compatibles}
```
Comment (Francisco2002) [inline dashboard/src/pages/hardwareDetails/HardwareDetails.tsx:231]: This expression can return undefined, but you define `compatibles` props in `CompatibleHardware` component as `string[]`.

```suggestion
            compatibles={data?.compatibles ?? []}
```

Comment (murilx) [reply dashboard/src/pages/hardwareDetails/HardwareDetails.tsx:231]: Actually, I believe the error I committed here was to add the `?` after data. data is already being tested for `undefined` or not being "prepared" above in the component so there's no need to add this `?` here, as far as I can tell.

**Review by WilsonNet — APPROVED** [review]
nice job

**Review by Francisco2002 — APPROVED** [review]
Nice job!!!!!!

**Pull request merged**